internal/components/setup/kind.go

```diff
@@ -73,8 +77,38 @@ type kindPort struct {
 	waitExpose string // Need to use when expose
 }
 
-//nolint:gocyclo // skip the cyclomatic complexity check here
+// pullImages pull docker image from a private repository
+func pullImages(images []string) error {
+	cli, err := docker.NewClientWithOpts(docker.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	var count int32
+	var wg sync.WaitGroup
+	for _, image := range images {
+		wg.Add(1)
+		go func(image string) {
+			defer wg.Done()
+			out, err := cli.ImagePull(context.Background(), image, types.ImagePullOptions{})
+			if err != nil {
+				logger.Log.Error("pull image error", "name", image, "error", err)
+			}
+			atomic.AddInt32(&count, 1)
+			out.Close()
+		}(image)
+	}
+	wg.Wait()
+	if int(count) != len(images) {
```

Comment:
I think this condition is always `true`? Because even if the error is detected, the count is always added. Please re-check and run it locally.